pkg/apis/openstack/validation/shoot.go

```diff
@@ -114,8 +117,39 @@ func ValidateWorkers(workers []core.Worker, cloudProfileCfg *api.CloudProfileCon
 	return allErrs
 }
 
+// validateWorkerZones validates worker zones for duplicates and existence in specified region
+func validateWorkerZones(zones []string, regionName string, regions []gardencorev1beta1.Region, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	var region *gardencorev1beta1.Region
+	for i := range regions {
+		if regionName == regions[i].Name {
+			region = &regions[i]
+			break
+		}
+	}
+	if region != nil {
+		usedZones := sets.NewString()
+	outer:
+		for i, zone := range zones {
+			if usedZones.Has(zone) {
+				allErrs = append(allErrs, field.Duplicate(fldPath.Index(i), zone))
+			}
+			usedZones.Insert(zone)
+			for _, z := range region.Zones {
+				if z.Name == zone {
+					continue outer
+				}
+			}
+			allErrs = append(allErrs, field.Invalid(fldPath.Index(i), zone, fmt.Sprintf("zone %s not existing in region %s", zone, region.Name)))
```

Comment:
I think that technically this check is already enforced on gardener level by the g/g admission checks.

Reply:
Ah, ok. But duplicate zones are missing at the moment. I will take a look in the g/g admission checks. Thanks!